### PR TITLE
Add words_to_remove for simple forms 21-0966

### PR DESCRIPTION
**Pull request by Thrillberg**
## Summary
This PR more accurately filters out PII for form 21-0966 error messages.

## Related issue(s)
https://app.zenhub.com/workspaces/vagov-product-team-forms-634853151f5c6000165942bc/issues/gh/department-of-veterans-affairs/va.gov-team/81261

### CODE REVIEW

File: modules/simple_forms_api/app/models/simple_forms_api/vba_21_0966.rb [hunk 1]
```diff
@@ -93,6 +99,82 @@ def track_user_identity(confirmation_number)
...
+      [
+        data['surviving_dependent_phone']&.gsub('-', '')&.[](0..2),
+        data['surviving_dependent_phone']&.gsub('-', '')&.[](3..5),
+        data['surviving_dependent_phone']&.gsub('-', '')&.[](6..9)
+      ]
+    end
+
+    def surviving_dependent_email
+      [
+        data['surviving_dependent_email']&.[](20..29),
```
Comment (rhasselle-oddball) [inline modules/simple_forms_api/app/models/simple_forms_api/vba_21_0966.rb:165]: just double checking, are these indices right?

**Review by Japenner — APPROVED** [review]
Looks good, great work! 👍

File: modules/simple_forms_api/app/models/simple_forms_api/vba_21_0966.rb [hunk 2]
```diff
@@ -93,6 +99,82 @@ def track_user_identity(confirmation_number)
 
     private
 
+    def veteran_ssn
+      [
+        data.dig('veteran_id', 'ssn')&.[](0..2),
+        data.dig('veteran_id', 'ssn')&.[](3..4),
+        data.dig('veteran_id', 'ssn')&.[](5..8)
```
Comment (rhasselle-oddball) [inline modules/simple_forms_api/app/models/simple_forms_api/vba_21_0966.rb:106]: do we also want to check for 'veteran_id', 'va_file_number' ?

File: modules/simple_forms_api/app/models/simple_forms_api/vba_21_0966.rb [hunk 3]
```diff
@@ -93,6 +99,82 @@ def track_user_identity(confirmation_number)
...
+        data['veteran_email']&.[](20..29),
+        data['veteran_email']&.[](0..19)
+      ]
+    end
+
+    def surviving_dependent_ssn
+      [
+        data.dig('surviving_dependent_id', 'ssn')&.[](0..2),
+        data.dig('surviving_dependent_id', 'ssn')&.[](3..4),
+        data.dig('surviving_dependent_id', 'ssn')&.[](5..8)
```
Comment (rhasselle-oddball) [inline modules/simple_forms_api/app/models/simple_forms_api/vba_21_0966.rb:144]: same

File: modules/simple_forms_api/app/models/simple_forms_api/vba_21_0966.rb [hunk 4]
```diff
@@ -93,6 +99,82 @@ def track_user_identity(confirmation_number)
...
+        data.dig('veteran_mailing_address', 'postal_code')&.[](0..4),
+        data.dig('veteran_mailing_address', 'postal_code')&.[](5..8)
+      ]
+    end
+
+    def veteran_home_phone
+      [
+        data['veteran_phone']&.gsub('-', '')&.[](0..2),
+        data['veteran_phone']&.gsub('-', '')&.[](3..5),
+        data['veteran_phone']&.gsub('-', '')&.[](6..9)
```
Comment (rhasselle-oddball) [inline modules/simple_forms_api/app/models/simple_forms_api/vba_21_0966.rb:129]: do we also want to include 'veteran_international_phone'?

File: modules/simple_forms_api/app/models/simple_forms_api/vba_21_0966.rb [hunk 5]
```diff
@@ -93,6 +99,82 @@ def track_user_identity(confirmation_number)
...
+        data.dig('surviving_dependent_mailing_address', 'postal_code')&.[](0..4),
+        data.dig('surviving_dependent_mailing_address', 'postal_code')&.[](5..8)
+      ]
+    end
+
+    def surviving_dependent_phone
+      [
+        data['surviving_dependent_phone']&.gsub('-', '')&.[](0..2),
+        data['surviving_dependent_phone']&.gsub('-', '')&.[](3..5),
+        data['surviving_dependent_phone']&.gsub('-', '')&.[](6..9)
```
Comment (rhasselle-oddball) [inline modules/simple_forms_api/app/models/simple_forms_api/vba_21_0966.rb:159]: same

**Review by rhasselle-oddball — COMMENTED** [review]
looks good, a couple of questions

File: modules/simple_forms_api/app/models/simple_forms_api/vba_21_0966.rb (hunk 1 above)
Comment (Thrillberg) [reply modules/simple_forms_api/app/models/simple_forms_api/vba_21_0966.rb:165]: Yep, matches the ones in the corresponding form_mappings file.

File: modules/simple_forms_api/app/models/simple_forms_api/vba_21_0966.rb (hunk 2 above)
Comment (Thrillberg) [reply modules/simple_forms_api/app/models/simple_forms_api/vba_21_0966.rb:106]: This field isn't split up the way the ssn field is so it'll get caught in our regular sweep for user input. The changes in this PR are only for fields that are split up (and therefore don't get caught when we just remove user-input directly).

File: modules/simple_forms_api/app/models/simple_forms_api/vba_21_0966.rb (hunk 4 above)
Comment (Thrillberg) [reply modules/simple_forms_api/app/models/simple_forms_api/vba_21_0966.rb:129]: Same reasoning as va file number, it gets plopped in wholesale so we don't need to specifically call it out here.

**Review by pennja — APPROVED** [review]
Looks good, great work! 👍

**Review by rhasselle-oddball — APPROVED** [review]

**Pull request merged**